### PR TITLE
Peanutposting on the importent bits of the supermatter

### PR DESCRIPTION
**Pull request by LemonInTheDark**
## About The Pull Request

Comments all of process_atmos() and some of the related static variables.
Adds some parenthesizes to make an if statement more clear.
## Why It's Good For The Game
This makes it easier to change the sm code without setting off a rube-goldberg machine, and given the talk of changing the engine, that sounds like an upside to me.
## Changelog
:cl:
code: Documented all of the supermatters's function process_atmos(). Check it out, it's dope
/:cl:

### CODE REVIEW

**Comment by LemonInTheDark** [comment]
Not sure what to do about the dme file, I can remove the changed line if you'd like.

**Comment by LemonInTheDark** [comment]
Also, we stan taking 2 weeks to make a comment pr

File: code/modules/power/supermatter/supermatter.dm [hunk 1]
```diff
@@ -748,15 +796,15 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_crystal)
 	layer = ABOVE_MOB_LAYER
 	moveable = TRUE
 
-/obj/machinery/power/supermatter_crystal/shard/engine
+	/obj/machinery/power/supermatter_crystal/shard/engine
```
Comment (bobbahbrown) [inline code/modules/power/supermatter/supermatter.dm]:
```suggestion
/obj/machinery/power/supermatter_crystal/shard/engine
```

File: code/modules/power/supermatter/supermatter.dm [hunk 2]
```diff
@@ -730,7 +778,7 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_crystal)
 				"<span class='danger'>The unearthly ringing subsides and you notice you have new radiation burns.</span>", MSG_AUDIBLE)
 		else
 			L.show_message("<span class='hear'>You hear an unearthly ringing and notice your skin is covered in fresh radiation burns.</span>", MSG_AUDIBLE)
-//Do not blow up our internal radio
+	//Do not blow up our internal radio
```
Comment (bobbahbrown) [inline code/modules/power/supermatter/supermatter.dm]:
```suggestion
//Do not blow up our internal radio
```

File: code/modules/power/supermatter/supermatter.dm [hunk 3]
```diff
@@ -748,15 +796,15 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_crystal)
...
 	is_main_engine = TRUE
 	anchored = TRUE
 	moveable = FALSE
 
-// When you wanna make a supermatter shard for the dramatic effect, but
-// don't want it exploding suddenly
-/obj/machinery/power/supermatter_crystal/shard/hugbox
+	// When you wanna make a supermatter shard for the dramatic effect, but
+	// don't want it exploding suddenly
+	/obj/machinery/power/supermatter_crystal/shard/hugbox
```
Comment (bobbahbrown) [inline code/modules/power/supermatter/supermatter.dm]:
```suggestion
/obj/machinery/power/supermatter_crystal/shard/hugbox
```

File: code/modules/power/supermatter/supermatter.dm [hunk 4]
```diff
@@ -748,15 +796,15 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_crystal)
...
 	name = "anchored supermatter shard"
 	is_main_engine = TRUE
 	anchored = TRUE
 	moveable = FALSE
 
-// When you wanna make a supermatter shard for the dramatic effect, but
-// don't want it exploding suddenly
-/obj/machinery/power/supermatter_crystal/shard/hugbox
+	// When you wanna make a supermatter shard for the dramatic effect, but
+	// don't want it exploding suddenly
```
Comment (bobbahbrown) [inline code/modules/power/supermatter/supermatter.dm]:
```suggestion
// don't want it exploding suddenly
```

File: code/modules/power/supermatter/supermatter.dm [hunk 5]
```diff
@@ -748,15 +796,15 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_crystal)
...
+	/obj/machinery/power/supermatter_crystal/shard/engine
 	name = "anchored supermatter shard"
 	is_main_engine = TRUE
 	anchored = TRUE
 	moveable = FALSE
 
-// When you wanna make a supermatter shard for the dramatic effect, but
-// don't want it exploding suddenly
-/obj/machinery/power/supermatter_crystal/shard/hugbox
+	// When you wanna make a supermatter shard for the dramatic effect, but
```
Comment (bobbahbrown) [inline code/modules/power/supermatter/supermatter.dm]:
```suggestion
// When you wanna make a supermatter shard for the dramatic effect, but
```

File: code/modules/power/supermatter/supermatter.dm [hunk 6]
```diff
@@ -488,13 +534,15 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_crystal)
 			supermatter_anomaly_gen(src, FLUX_ANOMALY, rand(5, 10))
 		if(power > SEVERE_POWER_PENALTY_THRESHOLD && prob(5) || prob(1))
 			supermatter_anomaly_gen(src, GRAVITATIONAL_ANOMALY, rand(5, 10))
-		if(power > SEVERE_POWER_PENALTY_THRESHOLD && prob(2) || prob(0.3) && power > POWER_PENALTY_THRESHOLD)
+		if((power > SEVERE_POWER_PENALTY_THRESHOLD && prob(2)) || (prob(0.3) && power > POWER_PENALTY_THRESHOLD))
```
Comment (bobbahbrown) [inline code/modules/power/supermatter/supermatter.dm]: this does clarify logic but order of operations would have evaluated this properly before
image

File: code/modules/power/supermatter/supermatter.dm [hunk 7]
```diff
@@ -441,19 +482,22 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_crystal)
 
 		//Also keep in mind we are only adding this temperature to (efficiency)% of the one tile the rock
 		//is on. An increase of 4*C @ 25% efficiency here results in an increase of 1*C / (#tilesincore) overall.
+		//Power * 0.55 * (some value between 1.5 and 15) / 5
 		removed.temperature += ((device_energy * dynamic_heat_modifier) / THERMAL_RELEASE_MODIFIER)
-
+		//We can only emmit so much heat, that being 37500
```
Comment (bobbahbrown) [inline code/modules/power/supermatter/supermatter.dm]:
```suggestion
		//We can only emit so much heat, that being 37500
```

File: code/modules/power/supermatter/supermatter.dm [hunk 8]
```diff
@@ -363,21 +366,42 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_crystal)
...
-			//capping damage
+			//caps heal rate
+
+			//Takes the larger between archived damage + (1.8) and damage
+			//This means we can only heal 1.8 damage per function call
 			damage = min(damage_archived + (DAMAGE_HARDCAP * explosion_point),damage)
 
 		removed.assert_gases(/datum/gas/oxygen, /datum/gas/plasma, /datum/gas/carbon_dioxide, /datum/gas/nitrous_oxide, /datum/gas/nitrogen, /datum/gas/pluoxium, /datum/gas/tritium, /datum/gas/bz)
 		//calculating gas related values
+		//Wanna� know� a� secret?� See� that� max()� to� zero?� it's� used� for� error� checking.� If� we� get� a� mol� count� in� the� negative,� we'll� get� a� divide� by� zero� error
```
Comment (bobbahbrown) [inline code/modules/power/supermatter/supermatter.dm]: what are these unicode characters

**Review by bobbah — CHANGES_REQUESTED** [review]
the weird indentation of ``/obj/machinery/power/supermatter_crystal/shard/engine`` and ``/obj/machinery/power/supermatter_crystal/shard/hugbox`` is absolutely haram

File: code/modules/power/supermatter/supermatter.dm (hunk 6 above)
Comment (LemonInTheDark) [reply code/modules/power/supermatter/supermatter.dm]: That's good, I'm glad to know it wasn't causing any issues.

File: code/modules/power/supermatter/supermatter.dm (hunk 8 above)
Comment (LemonInTheDark) [reply code/modules/power/supermatter/supermatter.dm]: You don't want to know.

File: code/modules/power/supermatter/supermatter.dm (hunk 1 above)
Comment (LemonInTheDark) [reply code/modules/power/supermatter/supermatter.dm]: Strong boomer energy, thanks for the pair of eyes

File: code/modules/power/supermatter/supermatter.dm (hunk 7 above)
Comment (LemonInTheDark) [reply code/modules/power/supermatter/supermatter.dm]: ... Thanks

Comment (LemonInTheDark) [reply code/modules/power/supermatter/supermatter.dm]: .... Thanks.

File: code/modules/power/supermatter/supermatter.dm (hunk 6 above)
Comment (optimumtact) [reply code/modules/power/supermatter/supermatter.dm]: people should get into the habit of using the brackets anyway

Comment (bobbahbrown) [reply code/modules/power/supermatter/supermatter.dm]: yeah i didnt mean to discourage it i just wanted to clarify that it didn't cause any bug which is what they said *might* be the case in their original pr sorry for the confusion folks

**Review by SpaceManiac — APPROVED** [review]
comments good

**Pull request merged**